test/test_core.py

```diff
@@ -3102,7 +3102,7 @@ def test_dlfcn_missing(self):
     if self.js_engines == [config.V8_ENGINE]:
       expected = "error: Could not load dynamic lib: libfoo.so\nError: Error reading file"
     else:
-      expected = "error: Could not load dynamic lib: libfoo.so\nError: ENOENT: no such file or directory, open 'libfoo.so'"
+      expected = "error: Could not load dynamic lib: libfoo.so\nError: ENOENT: no such file or directory"
```

Comment:
Is this change needed?  Seems like it would be good to check the filename appears in the error message.

Reply:
Without the change the test fails because the full path including the tmp dir appears in the error message, e.g.
```
 error: Could not load dynamic lib: libfoo.so
-Error: ENOENT: no such file or directory, open 'libfoo.so'
+Error: ENOENT: no such file or directory, open '/private/var/folders/wx/qv_075m16yx5lqwvjfwswfqm0000gn/T/emscripten_test_core0_jwfom28o/libfoo.so'
```

I considered using something like,
```
expected = "error: Could not load dynamic lib: libfoo.so\nError: ENOENT: no such file or directory, open '.*libfoo.so'"
```
with
```
self.do_run(src, expected, regex=True)
```
but decided against it. Would you prefer that?

Reply:
Perhaps we should use the original name in the error message then?

Reply:
On a second look, it seems the first line of the error already does include the original filename. Since the name also appears in the first part of the error message, perhaps that is enough?

I think the problem is that the exception `e` is directly added as the next line, and I'm not sure how much control there is over how that is printed.

https://github.com/emscripten-core/emscripten/blob/aab32efdcc9d1157fe925678ae5806a8f371d4ec/src/library_dylink.js#L997

Reply:
OK fair enough.